test/server/services/ApplicationDeveloper.test.js

```diff
@@ -202,6 +202,36 @@ describe('services/ApplicationDeveloper', () => {
     expect(stubOpts).to.have.been.calledOnce;
     expect(stubLogger).to.have.been.calledOnce;
   });
+  it('should save a rest key using the applications.api service', () => {
+    // static test vars
+    const mockEnv = 'mock';
+    const mockURL = 'https://mocksite.com';
+    const finalVars = Object.assign({}, baseOptions);
+    const mockId = 'test';
+    finalVars.method = 'POST';
+    finalVars.url = mockURL + '/applications/'+mockId+'/api-clients/key';
+    finalVars.data = {
+      description: 'description',
+      name: 'key demo',
+      restricted_ips: ['00000'],
+      js_domains: ['javascriptDomain']
+    };
```

Comment:
I might use `Object.assign({}, routeData.addNewKey)` so that the test should remain correct even if we need to update data in addNewKey.